src/cpp/core/include/core/Log.hpp

```diff
@@ -82,7 +82,9 @@ std::string errorAsLogEntry(const Error& error);
 #define LOG_INFO_MESSAGE_NAMED(logSection, message) rstudio::core::log::logInfoMessage(message, \
                                                                                        logSection)
 
-#define LOG_DEBUG_MESSAGE(message) rstudio::core::log::logDebugMessage(message)
+
+#define LOG_DEBUG_MESSAGE(message) (rstudio::core::log::isLogLevel(rstudio::core::log::LogLevel::DEBUG_LEVEL) ? rstudio::core::log::logDebugMessageReturn(message) : false)
```

Comment:
I would recommend adding some tests of this new macro as you mentioned - you tried several ways, would be good to see them used to make sure we support the style we want.

Reply:
Sure, good idea.

Reply:
Ok I added unit tests for the key cases.